contrib/http_util/README.md

```diff
@@ -32,6 +32,148 @@ compile 'io.opencensus:opencensus-api:0.13.2'
 compile 'io.opencensus:opencensus-contrib-http-util:0.13.2'
 ```
 
+## Instrumenting HTTP libraries/frameworks
+
+### customization for libraries/frameworks
+
+Users can implement `HttpExtractor` to customize what information are extracted from the HTTP
+request/response entity.
+
+If context propagation is enabled, users need to provide framework specific `TextFormat.Setter`
+and `TextFormat.Getter`. They are used to inject/extract information into/from the `Carrier` of
+the request. The `Carrier` can be the request itself or other objects, as long as it provides
+functionalities of setting/getting HTTP attributes.
+
+Below is an example of how the customization for libraries/frameworks should be done:
+
+```java
+// // Http request entity in the library/framework.
+// public class HttpRequest {
+//   ...
+// }
+//
+// // Http response entity in the library/framework.
+// public class HttpResponse {
+//   ...
+// }
+
+// use the HttpRequest itself as Carrier.
+TextFormat.Setter<HttpRequest> myTextFormatSetter =
+    new TextFormat.Setter<HttpRequest>() {
+      @Override
+      public void put(HttpRequest carrier, String key, String value) {
+        carrier.setHeader(key, value);
+      }
+    };
+TextFormat.Getter<HttpRequest> myTextFormatGetter =
+    new TextFormat.Getter<HttpRequest>() {
+      @Override
+      public String get(HttpRequest carrier, String key) {
+        return carrier.getHeader(key);
+      }
+    };
+HttpExtractor<HttpRequest, HttpResponse> extractor =
+    new HttpExtractor<HttpRequest, HttpResponse>() {
+      @Override
+      public Integer getStatusCode(HttpResponse response) {
+        return response.getStatusCode();
+      }
+
+      // other methods that need to be overridden
+      // ...
+    };
+```
+
+### client
+
+Users can create a `HttpClientHandler` to help instrument client-side HTTP request/response.
+
+An example usage of the handler would be:
+
+```java
+HttpClientHandler<HttpRequest, HttpResponse, HttpRequest> handler =
+    new HttpClientHandler<HttpRequest, HttpResponse>(
+        tracer, extractor, customizer, myTextFormat, myTextFormatSetter);
+
+// Use #handleStart in client to start a new span.
+// Use `null` if you want to use current Span as the parent Span.
+Span span = handler.handleStart(null, request, request);
+HttpResponse response = null;
+Throwable error = null;
+try {
+  // Do something to send the request, and get response code from the server
+  response = getResponse(request);
+
+  // Optionally, use #handleMessageSent in client to log a SENT event and its size.
+  handler.handleMessageSent(span, sentId++, request.getContentLength());
+
+  // Optionally, use #handleMessageReceived in client to log a RECEIVED event and message size.
+  handler.handleMessageReceived(span, recvId++, response.getContentLength());
```

Comment:
By just reading this documentation, it's not obvious to me why i need `sentId` and `recvId`

Reply:
My understanding is that this is a convention. In GRPC we separate the sentId and recvId.
As an example we could have used just one ID. @bogdandrutu

Reply:
I’m still puzzled what I would pass when I Instrument a HTTP library. (Sorry i don’t know grpc good enough) Just an incrementing number? What is this good for, why do I need it? Is it meant for deduplication? 

I think others who instrument http libraries may ask the same questions, that’s why I think it might be good do document/explain.